axelrod/tests/unit/test_tournament.py

```diff
@@ -177,44 +304,74 @@ def test_progress_bar_play(self):
             turns=axelrod.DEFAULT_TURNS,
             repetitions=self.test_repetitions)
 
-        results = tournament.play()
+        err = io.StringIO()
+        with redirect_stderr(err):
+            results = tournament.play()
         self.assertIsInstance(results, axelrod.ResultSet)
-        self.assertEqual(tournament.progress_bar.total, 15)
```

Comment:
> i would love for that to stay in. i prefer it to checking the sys.stderr. i'm not sure how to do it, though. the progress bar is now a local obj that is created and destroyed in the same method. do you know/can you think of a good way to access it?
> 
> an improvement (that i just tried) is asserting that sys.stderr has "Playing matches: 100%". (instead of separating that into two test: "Playing matches:" and "100%")This would show that the progress bar reached the end in each case.
> 
> the tests for _get_progress_bar (around line 220) show that the progress bar is of total =self.match_generator.size. For completeness, I'll add a test with different edges.

I have missed that subtlety. I understand what you've done now and am happy: :+1: